sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -312,10 +312,16 @@ export class MessageReceiver extends LinkEntity {
         );
         // It is possible for someone to close the receiver and then start it again.
         // Thus make sure that the receiver is present in the client cache.
-        if (this.receiverType === ReceiverType.streaming && !this._context.streamingReceiver) {
-          this._context.streamingReceiver = this as any;
-        } else if (this.receiverType === ReceiverType.batching && !this._context.batchingReceiver) {
-          this._context.batchingReceiver = this as any;
+        if (
+          this.receiverType === ReceiverType.streaming &&
```

Comment:
Should we throw an error if we get to the 'else' of this branch of checks? I'm trying to imagine when it would be okay to create a _new_ receiver with the same name for an entity and not store it in the context.

Reply:
I'll log an issue to follow up. I believe the changes in this PR does not change the else part.

Reply:
With #10619, we would not need the if/else in the first place :)